### PR TITLE
adding version to documentation to help out on #1016

### PR DESCRIPTION
**Pull request by studioj**
#1016 suggested a rename version feature.
although not implemented yet it seems possible with the Version.update functionality

This addition to the documentation will at least make it more clear where to look for such solutions

### CODE REVIEW

**Comment by studioj** [comment]
i see @adehad  is also fixing this (halfly) with this PR https://github.com/pycontribs/jira/pull/1023/files
although my implementation doesn't follow his changed doc generation

File: jira/resources.py [hunk 1]
```diff
@@ -899,7 +900,13 @@ def delete(self, moveFixIssuesTo=None, moveAffectedIssuesTo=None):
         return super(Version, self).delete(params)
 
     def update(self, **args):
-        """Update this project version from the server. It is prior used to archive versions."""
+        """
+        Update this project version from the server. It is prior used to archive versions.
+
+        link to atlassian restapi `documentation`_.
```
Comment (adehad) [inline jira/resources.py:906]: As this docstring will be user facing in the docstring, we can make it a little nicer ;)

```suggestion
        Refer to Atlassian  REST API `documentation`_ for all possible values.
```

File: jira/resources.py [hunk 2]
```diff
@@ -899,7 +900,13 @@ def delete(self, moveFixIssuesTo=None, moveAffectedIssuesTo=None):
...
+
+        .. _documentation: https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-project-versions/#api-rest-api-2-version-id-get
```
Comment (adehad) [inline jira/resources.py:908]:
```suggestion
        .. _documentation: https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-project-versions/#api-rest-api-2-version-id-put
```

The current link points to the `get` method, the suggested link points to the `put` method which is what is used when updating.

File: jira/resources.py [hunk 3]
```diff
@@ -899,7 +900,13 @@ def delete(self, moveFixIssuesTo=None, moveAffectedIssuesTo=None):
         return super(Version, self).delete(params)
 
     def update(self, **args):
```
Comment (adehad) [inline jira/resources.py:902]: it bothers me a little that this is `**args`, we would expect it to follow the convention of `**kwargs`. 

```suggestion
    def update(self, **kwargs):
```

This change would also require changes in the function to how it is used.

**Review by adehad — CHANGES_REQUESTED** [review]
As you say I've made some changes in my branch too, I'm happy for this to be merged first, as my PR is a bit large and probably will take a while to get approved. Until then I think let's get this merged, only a few minor comments from me.

File: jira/resources.py (hunk 3 above)
Comment (studioj) [reply jira/resources.py:902]: Well that would be a change in the public API which we should think twice to change...
I wouldn't do a change like this in this pr.

Maybe we should list up small improvements which might break the API and prepare for them

File: jira/resources.py (hunk 2 above)
Comment (studioj) [reply jira/resources.py:908]: thx for checking (y)

File: jira/resources.py (hunk 3 above)
Comment (adehad) [reply jira/resources.py:902]: Changing this variable name would not affect the public API. 

`**` indicates an unspecified number of **keyword** arguments. This variable is a catch all for these keyword arguments and it's name won't affect users in anyway.

Comment (studioj) [reply jira/resources.py:902]: darnit indeed you're right `*args ` and `**kwargs`

Comment (studioj) [reply jira/resources.py:902]: 🤔 this change is not tested. I'll add a test for it too

**Review by adehad — APPROVED** [review]
Excellent

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**